dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -0,0 +1,239 @@
+  };
+  const saveFavorites = (fav) => {
+    localStorage.setItem("favControllers", JSON.stringify(fav));
+  };
+  const Sidebar = (
+    <PageSidebar
+      nav={<NavItems />}
+      className="sidebar"
+      isNavOpen={isNavOpen}
+    />
+  );
+  const NavbarDrawer = () => {
+    return (
+      <Masthead id="basic">
+        <MastheadToggle>
+          <PageToggleButton isNavOpen={isNavOpen} onNavToggle={onNavToggle}>
+            <BarsIcon />
+          </PageToggleButton>
+        </MastheadToggle>
+      </Masthead>
+    );
+  };
+  const selectedArray =
+    isSelected === "firstButton" ? publicData : favoriteRepoNames;
+  return (
+    <>
+      <Page header={<NavbarDrawer />} sidebar={Sidebar}>
+        <AlertMessage
+          message="Want to see only metric relevant to you?"
+          link="Login to create an account"
+        />
+        <PageSection variant={PageSectionVariants.light}>
+          <PathBreadCrumb pathList={["Dashboard", "Components"]} />
+          <Heading headingTitle="Controllers"></Heading>
+          <div className="filterContainer">
+            <SearchBox
+              dataArray={dataArray}
+              setPublicData={setPublicData}
+              startDate={startDate}
+              endDate={endDate}
+              setControllerName={setControllerName}
+            />
+            <DatePickerWidget
+              dataArray={dataArray}
+              setPublicData={setPublicData}
+              controllerName={controllerName}
+              setDateRange={setDateRange}
+            />
+          </div>
+          <ToggleGroup aria-label="Available options with Single Selectable">
+            <ToggleGroupItem
+              text={`All Controllers(${publicData.length})`}
+              buttonId="firstButton"
+              isSelected={isSelected === "firstButton"}
+              onChange={handleButtonClick}
+              className="controllerListButton"
+            />
+            <ToggleGroupItem
+              text={`Favorites(${favoriteRepoNames.length})`}
+              buttonId="secondButton"
+              isSelected={isSelected === "secondButton"}
+              onChange={handleButtonClick}
+              className="favoriteListButton"
+            />
+          </ToggleGroup>
+          <TableComposable aria-label="Favoritable table" variant="compact">
+            <Thead>
+              <Tr>
+                <Th sort={getSortParams(0)}>{columnNames.controller}</Th>
+                <Th sort={getSortParams(1)}>{columnNames.name}</Th>
+                <Th sort={getSortParams(2)}>{columnNames.creationDate}</Th>
+                <Th sort={getSortParams(3)}></Th>
+              </Tr>
+            </Thead>
+            <Tbody>
+              {selectedArray.length > 0 ? (
+                selectedArray.map((repo, rowIndex) => (
+                  <Tr key={rowIndex}>
+                    <Td dataLabel={columnNames.controller}>
+                      <a href="#">{repo.controller}</a>
```

Comment:
The plan is that if a user clicks on any particular controller,he will be directed to a new page where he can see table of content or maybe even do data comparison.Since we dont have such page as of now,I put "#" symbol as of now.

Reply:
In the 0.69 model, "controller" is a grouping for multiple datasets; so clicking on a controller in the main landing page would take you to a list of datasets using that controller name. Clicking on one of those controllers would take you to some sort of a "first level" visualization. I think we need to decide exactly what that visualization will be. Possibly even a split view with a TOC panel and a panel of system configuration information, tool summary ... I'm not sure at this point.

But remember that for 0.72 and beyond, it's the username that replaces the old grouping construct of "controller", and the browsing page should really be focused on the datasets the browsing user is allowed to read. (Subject to various filtering criteria, probably including controller as well as date range, benchmark name, etc.) I don't see any obvious (or likely useful) "first level" default grouping here; and although we may come up with something, it won't be "controller"!

Reply:
In the future, it would be good to mark things like this with a `TODO` comment, so that the reader (and future developers) know that the code as presented is incomplete.  Placeholders are generally fine, so long as it's clear that they are placeholders, and so long as they are easy to locate later.  (We can quibble about whether `"#"` is obviously a placeholder, but a comment would make it clear.)